DQM/SiPixelPhase1Summary/src/SiPixelPhase1Summary.cc

```diff
@@ -325,20 +328,40 @@ void SiPixelPhase1Summary::fillSummaries(DQMStore::IBooker& iBooker, DQMStore::I
       }
       for (int j = 0; j < 4; j++) {  // !??!?!? yAxisLabels_.size() ?!?!?!
         //Ignore the bins without detectors in them
-        if (i == 1 && j > 1)
-          continue;
-        summaryMap_["Grand"]->setBinContent(
-            i + 1,
-            j + 1,
-            1);  // This resets the map to be good. We only then set it to 0 if there has been a problem in one of the other summaries.
-        if (deadROCSummary->getBinContent(i + 1, j + 1) > deadRocThresholds_[i * 4 + j])
-          summaryMap_["Grand"]->setBinContent(i + 1, j + 1, 0);
-        sumOfNonNegBins += summaryMap_["Grand"]->getBinContent(i + 1, j + 1);
+        if (i == 1 && j > 1) {
+          summaryMap_["Grand"]->setBinContent(i + 1, j + 1, -1);
+        } else {
+          if (deadROCSummary->getBinContent(i + 1, j + 1) < deadRocWarnThresholds_[i * 4 + j])
+            summaryMap_["Grand"]->setBinContent(i + 1, j + 1, 1);
+
+          else if (deadROCSummary->getBinContent(i + 1, j + 1) > deadRocWarnThresholds_[i * 4 + j] &&
+                   deadROCSummary->getBinContent(i + 1, j + 1) < deadRocThresholds_[i * 4 + j])
+            summaryMap_["Grand"]->setBinContent(i + 1, j + 1, 0.8);
+
+          else
+            summaryMap_["Grand"]->setBinContent(i + 1, j + 1, 0);
+
+          sumOfNonNegBins += summaryMap_["Grand"]->getBinContent(i + 1, j + 1);
+        }
+        if (i == 1 && j > 1) {
```

Comment:
I miss why the very same code has to be repeated twice (L331-345 vs L346-360)

Reply:
Hi @perrotta, My apologies! I think somehow these lines had been added twice. Is there any way to remove L346-360 from here or do I have to push it again?

Reply:
Thank you @sayanchatterjee38 
I think that pushing a new commit is the easiest way to proceed.
Only if you are at ease with it, you could further squeeze all commits into a single one afterwards